matchbox_signaling/src/signaling_server/builder.rs

```diff
@@ -114,22 +114,33 @@ where
         self
     }
 
-    /// Create a [`SignalingServer`].
-    pub fn build(self) -> SignalingServer {
+    /// Construct the router to use directly
+    ///
+    /// <div class="warning">
+    /// You must call `into_make_service_with_connect_info::<SocketAddr>()` before serving this
+    /// router for matchbox to work
+    /// </div>
```

Comment:
This is a little concerning. It seems too easy to get wrong. Perhaps a flyway pattern could help us?

However, I'd rather pass the 2 routes (`"/"` and `"/:path"`) as arguments to `build` or have some `build_with(Router)` method alternative.

Reply:
Flyway pattern?

Yeah I was hoping that there was some way of passing across the service that ensured this but couldn't find a way.
A `build_with` alternative sounds like a good idea though, I think it's best to have something more flexible than just passing routes.

Reply:
I've implemented a `build_with` alternative, and added a test for how it can be used too.

Reply:
Bump for a review when you get a chance